linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -135,6 +136,24 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
     }
   }
 
+  private def checkJSSuperClass(classDef: LinkedClass): Unit = {
+    implicit val ctx = ErrorContext(classDef)
+
+    val hasJSSuper = classDef.jsSuperClass.isDefined
+
+    if (classDef.kind.isJSClass) {
+      if (!hasJSSuper) {
+        // .get is OK: the Analyzer checks that a super class is present.
+        val superClass = lookupClass(classDef.superClass.get.name)
+        if (superClass.jsClassCaptures.isDefined)
+          reportError(i"super class ${superClass.name} may not have jsClassCaptures")
+      }
```

Comment:
Shouldn't we have an `else` branch that typechecks the `Tree` in the `jsSuperClass`, while we're here?